src/ArborX_LinearBVH.hpp

```diff
@@ -227,19 +230,26 @@ class BoundingVolumeHierarchy
   }
 };
 
-template <typename MemorySpace>
-using BVH = BoundingVolumeHierarchy<MemorySpace>;
+template <
+    typename MemorySpace, typename Value = Details::LegacyDefaultTemplateValue,
+    typename IndexableGetter = Details::DefaultIndexableGetter,
+    typename BoundingVolume = ExperimentalHyperGeometry::Box<
+        GeometryTraits::dimension_v<
+            std::decay_t<std::invoke_result_t<IndexableGetter, Value>>>,
+        typename GeometryTraits::coordinate_type<
+            std::decay_t<std::invoke_result_t<IndexableGetter, Value>>>::type>>
```

Comment:
Three strikes you're out...
Define a helper trait that takes IndexableGetter and Value and deduces a bounding_volume_type.

Reply:
This is not a trait. It's not possible to deduce the bounding volume type from indexable getter and a value. How about instead I do
```
template <typename IndexableGetter, typename Value>
using BoxFromIndexable = ExperimentalHyperGeometry::Box<
    GeometryTraits::dimension_v<
        std::decay_t<std::invoke_result_t<IndexableGetter, Value>>>,
    typename GeometryTraits::coordinate_type<
        std::decay_t<std::invoke_result_t<IndexableGetter, Value>>>::type>;
```

Reply:
This is more or less what I was trying to suggest.

Reply:
It is not obvious where that trait would sit so maybe we close our eyes until we know better.